tests/testsrc/tests/js1_7/geniter/regress-352197.js

```diff
@@ -21,7 +21,7 @@ function test()
   printBugNumber(BUGNUMBER);
   printStatus (summary);
 
-  expect = /TypeError: anonymous generator function returns a value/;
+  expect = /SyntaxError: anonymous generator function returns a value/;
```

Comment:
Apart from the ECMA specifications, to make the code easier, I also changed the old generator error.